js/samples/ios/quickstart-swift/ImmersiveReader.swift

```diff
@@ -3,44 +3,94 @@
 
 import Foundation
 
-public struct Content: Encodable {
+@objc public class Content: NSObject, Encodable {
     var title: String
     var chunks: [Chunk]
 
-    public init(title: String, chunks: [Chunk]) {
+    @objc public init(title: String, chunks: [Chunk]) {
         self.title = title
         self.chunks = chunks
     }
 }
 
-public struct Chunk: Encodable {
+@objc public class Chunk: NSObject, Encodable {
     var content: String
     var lang: String?
     var mimeType: String?
 
-    public init(content: String, lang: String?, mimeType: String?) {
+    @objc public init(content: String, lang: String?, mimeType: String?) {
         self.content = content
         self.lang = lang
         self.mimeType = mimeType
     }
 }
 
-public struct Options: Encodable {
+@objc public class Options: NSObject, Encodable {
     var uiLang: String?
-    var timeout: TimeInterval?
+    var timeout: Int?
     var uiZIndex: Int?
     var useWebview: Bool?
     var customDomain: String?
     var allowFullscreen: Bool?
+    var parent: String?
```

Comment:
>String [](start = 16, length = 6)

in the js sdk, this is a dom element, might want to remove option here to prevent potential confusion for ios devs